tests/RobotFramework/tests/cumulocity/monitoring/monitor_device_collectd.robot

```diff
@@ -0,0 +1,69 @@
+*** Settings ***
+Documentation      With thin-edge.io device monitoring, you can collect metrics from your device 
+...                and forward these device metrics to IoT platforms in the cloud. Using these metrics, 
+...                you can monitor the health of devices and can proactively initiate actions in case the 
+...                device seems to malfunction. Additionally, the metrics can be used to help the customer 
+...                troubleshoot when problems with the device are reported. Thin-edge.io uses the open source 
+...                component collectd to collect the metrics from the device. Thin-edge.io translates the 
+...                collected metrics from their native format to the thin-edge.io JSON format and then into 
+...                the cloud-vendor specific format.
+...                Enabling monitoring on your device is a 3-steps process:
+...                1. Install collectd,
+...                2. Configure collectd,
+...                3. Enable thin-edge.io monitoring.
+...                Checking the grouping of measurements:
+...                Sending fake collectd measurements with stopped collectd and check that messages are
+...                published on tedge/measurements for that timestamp and grouping the two fake collectd measurements 
+...                (i.e. the temperature and the pressure sent by the two fake measurements).
+
+
+Resource        ../../../resources/common.resource
+Library         ThinEdgeIO
+Library         Cumulocity
+
+Suite Setup     Custom Setup
+Test Teardown    Get Logs
+
+Force Tags      theme:monitoring    theme:c8y    theme:collectd
+
+*** Test Cases ***
+
+Check running collectd
+    Service Should Be Running    collectd
+    
+Is collectd publishing MQTT messages?
+    ${messages}=    Should Have MQTT Messages    topic=collectd/#    minimum=1    maximum=None
+
+Check thin-edge monitoring
+    # 3. Enable thin-edge.io monitoring.
+    Execute Command    sudo systemctl enable tedge-mapper-collectd
+    Execute Command    sudo systemctl start tedge-mapper-collectd
+    # Check thin-edge monitoring
+    ${tedge_messages}=    Should Have MQTT Messages    topic=tedge/measurements    minimum=1    maximum=None
+    Should Contain    @{tedge_messages}   "time"
+    Should Contain Any    @{tedge_messages}    "memory"    "cpu"    "df-root"
+    ${c8y_messages}=    Should Have MQTT Messages    topic=c8y/measurement/measurements/create    minimum=1    maximum=None
+    Should Contain    @{c8y_messages}    "type":"ThinEdgeMeasurement"
```

Comment:
```suggestion
    Should Contain    @{c8y_messages}    "type":"ThinEdgeMeasurement"
    Should Contain Any    @{c8y_messages}    "memory"    "cpu"    "df-root"
```
The precise validation would be to make sure that the same measurement key(`cpu`, `memory` or `df-root`) that was present in the `tedge_message` is present in the corresponding `c8y_message` as well. But, that might be a bit harder as there could be many messages on `tedge_messages`. So, the `Any` validation would be fine for now.